doc/_scripts/software_maturity_scanner.py

```diff
@@ -0,0 +1,554 @@
+    rule -> or-rule
+    or-rule -> and-rule | and-rule '||' or-rule
+    and-rule -> sub-rule | sub-rule '&&' and-rule
+    sub-rule -> symbol | '(' or-rule ')'
+    symbol -> any matches of the regular expression '[A-Z0-9_]+'
+
+    And-expressions have a higher precedence than or-expressions, making the
+    rule 'A || B && C' equivalent with 'A || (B && C)'
+
+    Each list in the returned nested list of expressions is prefixed by the
+    'AND' or 'OR' keyword.
+
+    For example, the rule 'A && (B || C)' will return the following list:
+    [OR, [AND, A, [OR, [AND, B], [AND, C]]]]
+    """
+
+    and_expressions = split_on("||", rule)
+    parsed_rule = ["OR"]
+    for expression in and_expressions:
+        symbols = split_on("&&", expression)
+        for i, symbol in enumerate(symbols):
+            if symbol.startswith("(") and symbol.endswith(")"):
+                symbols[i] = parse_rule(symbol[1:-1])
+            elif not re.match(r"^[A-Z0-9_]+$", symbol):
+                logger.error(f"Invalid Kconfig symbol '{symbol}'")
+            elif not symbol.startswith("CONFIG_"):
+                symbols[i] = "CONFIG_" + symbol
+        symbols.insert(0, "AND")
+        parsed_rule.append(symbols)
+    return parsed_rule
+
+
+def evaluate_rule(vars: Dict[str, kconfiglib.Variable], rule: list) -> bool:
+    """Evaluate a parsed Kconfig rule against a dictionary of Kconfig variables.
+
+    Returns:
+        False if any of the required variables are missing, True otherwise
+    """
+
+    assert rule[0] == "OR"
+    # At least one of the and-rules must evaluate to True
+    for and_rule in rule[1:]:
+        assert and_rule[0] == "AND"
+        # All of the symbols in the and-rule must be present
+        for symbol in and_rule[1:]:
+            # A nested list indicates a subrule in parentheses
+            if isinstance(symbol, list):
+                if not evaluate_rule(vars, symbol):
+                    break
+            # The and-rule fails if a symbol is not present
+            elif symbol not in vars or vars[symbol].value != "y":
+                break
+        # No breaks in for-loop means all symbols are present
+        else:
+            return True
+    return False
+
+
+def find_maturity_level(rule: list, experimental_symbols: Set[str]) -> str:
+    """Find if any of the symbols used in a rule are experimental or not.
+
+    Returns:
+        'Experimental' | 'Supported'
+    """
+
+    def flatten(items):
+        """Flatten the rule list to a set of unique symbols."""
+        out = set()
+        for item in items:
+            if isinstance(item, list):
+                out.update(flatten(item))
+            else:
+                out.add(item)
+        return out
+
+    symbols = flatten(rule)
+    for symbol in symbols:
+        if symbol in experimental_symbols:
+            return "Experimental"
+    return "Supported"
+
+
+def generate_tables(args) -> None:
```

Comment:
I would prefer to use args only in `__name__ == '__main__'` block. Because, in theory, other methods from script can be run from other python packages. Please change method api to `generate_tables(build_output, input_file, ...)`

Reply:
All command line arguments are now named arguments in `generate_tables`